### PR TITLE
Document and test HiveFileSplit

### PR DESCRIPTION
**Pull request by elharo**
## Description
Added tests and API doc for HiveFileSplit

## Motivation and Context
"split" does not seem to be a common term of the art in computer science. The only references I could find to it referred to splitting ML data sets into training and verification data, which is not what this is. Also "split" is very hard to google for, and zero is not a positive number.

## Impact
More accurate precondition violation messages.

## Test Plan
CI

## Contributor checklist

- [x] Please make sure your submission complies with our [development](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#development), [formatting](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#formatting), [commit message](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#commit-formatting-and-pull-requests), and [attribution guidelines](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#attribution).
- [x] PR description addresses the issue accurately and concisely.  If the change is non-trivial, a GitHub Issue is referenced.
- [x] Documented new properties (with its default value), SQL syntax, functions, or other functionality.
- [ ] If release notes are required, they follow the [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines).
- [x] Adequate tests were added if applicable.
- [ ] CI passed.

## Release Notes

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

**Review by steveburnett — APPROVED** [review]
LGTM! (docs)

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveFileSplit.java [hunk 1]
```diff
@@ -24,6 +24,11 @@
 import static com.google.common.base.Preconditions.checkArgument;
 import static java.util.Objects.requireNonNull;
 
+/**
+ * Represents a piece of a file at a given path starting at a given position and continuing for length bytes.
+ * All numeric values must be zero or greater. Object values must be non-null.
```
Comment (rschlussel) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveFileSplit.java:29]: Suggest removing this line. I found it hard to understand what you meant here (i thought you were saying all numeric values in the file had to be non-negative), it can easily get out of date if someone happens to add some other field that can be negative, and this information is pretty easy to find in the constructor.  If you don't want to remove it I suggest specifying which fields can't be negative.  
```suggestion
```

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveFileSplit.java [hunk 2]
```diff
@@ -24,6 +24,11 @@
...
+ * There is no guarantee the file exists, or, if it exists, that it is large enough to provide length bytes.
```
Comment (rschlussel) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveFileSplit.java:30]: I would remove this as well. I guess technically, there's no guarantee about these things because this is just a data structure storing some information, but we do get the path from the metastore and then get metadata for the file from the file system to be able to populate this, so if we exclude concurrent updates, then we do have guarantees about this.

**Review by rschlussel — COMMENTED** [review]
Would be nice to add documentation to the ConnectorSplit interface as well to explain that a split represents a chunk of work for each worker to process.

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveFileSplit.java (hunk 2 above)
Comment (elharo) [reply presto-hive/src/main/java/com/facebook/presto/hive/HiveFileSplit.java]: The API doc for a public class should not assume the class will only be used in a particular way. Unless we can make this class non-public (which maybe we can do? I've noticed Presto tends to have more public API than it needs.) This class provides no such guarantees.

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveFileSplit.java (hunk 1 above)
Comment (elharo) [reply presto-hive/src/main/java/com/facebook/presto/hive/HiveFileSplit.java]: done

**Comment by elharo** [comment]
> Would be nice to add documentation to the ConnectorSplit interface as well to explain that a split represents a chunk of work for each worker to process.

https://github.com/prestodb/presto/issues/22458

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveFileSplit.java (hunk 2 above)
Comment (rschlussel) [reply presto-hive/src/main/java/com/facebook/presto/hive/HiveFileSplit.java]: maybe rephrase it to say that the creator of a HiveFileSplit is responsible for ensuring the existence of the file, start and length, and this class does not do any such validation itself.

Comment (elharo) [reply presto-hive/src/main/java/com/facebook/presto/hive/HiveFileSplit.java]: rephrased

**Review by rschlussel — APPROVED** [review]
